### PR TITLE
Fixes beauty runtimes and accidental value changes

### PR DESCRIPTION
**Pull request by ghost**
:cl: Qustinnus
fix: beauty no longer runtimes like crazy
fix: glass and silver are back to normal values after accidentally changing them
/:cl:

get_area doesnt work on things inside of things apparently.

### CODE REVIEW

File: code/datums/materials/basemats.dm [hunk 1]
```diff
@@ -19,7 +19,6 @@
 	integrity_modifier = 0.1
 	sheet_type = /obj/item/stack/sheet/glass
 	value_per_unit = 0.001
-	beauty_modifier = 0.05
```
Comment (Fikou) [inline code/datums/materials/basemats.dm]: undocumented change?!!?!!?!!

Comment (Qustinnus) [reply code/datums/materials/basemats.dm]: fixed

File: code/datums/components/beauty.dm [hunk 2]
```diff
@@ -11,13 +11,13 @@
 	enter_area(null, A)
 
 /datum/component/beauty/proc/enter_area(datum/source, area/A)
-	if(A.outdoors)
+	if(!istype(A) || A.outdoors)
```
Comment (ninjanomnom) [inline code/datums/components/beauty.dm]: Don't istype here, check if A exists before your non signal calls on this proc. The signal version will always have the proper area provided

File: code/datums/components/beauty.dm [hunk 3]
```diff
@@ -11,13 +11,13 @@
...
 		return
 	A.totalbeauty += beauty
 	A.update_beauty()
 
 /datum/component/beauty/proc/exit_area(datum/source, area/A)
-	if(A.outdoors)
+	if(!istype(A) || A.outdoors)
```
Comment (ninjanomnom) [inline code/datums/components/beauty.dm]: Ditto

**Review by ninjanomnom — CHANGES_REQUESTED** [review]

File: code/datums/materials/basemats.dm (hunk 1 above)
Comment (ninjanomnom) [reply code/datums/materials/basemats.dm]: not fixed

File: code/datums/components/beauty.dm [hunk 4]
```diff
@@ -8,7 +8,8 @@
 	RegisterSignal(parent, COMSIG_ENTER_AREA, .proc/enter_area)
 	RegisterSignal(parent, COMSIG_EXIT_AREA, .proc/exit_area)
 	var/area/A = get_area(parent)
-	enter_area(null, A)
+	if(istype(A))
```
Comment (ninjanomnom) [inline code/datums/components/beauty.dm]:
```suggestion
	if(A) 
```

File: code/datums/components/beauty.dm [hunk 5]
```diff
@@ -25,4 +26,5 @@
 /datum/component/beauty/Destroy()
 	. = ..()
 	var/area/A = get_area(parent)
-	exit_area(null, A)
+	if(istype(A))
```
Comment (ninjanomnom) [inline code/datums/components/beauty.dm]:
```suggestion
	if(A)
```

**Review by ninjanomnom — CHANGES_REQUESTED** [review]

**Pull request merged**